### ISSUE TITLE
Weird [?2004h and [?2004l, and problem with initialization of graphics_toolkit on Manjaro and Arch Linux

### ISSUE THREAD

**Issue by ThomasUCAS** [Calysto/octave_kernel#218]
The environment is set as "pacman -S octave" and "conda install -c conda-forge octave_kernel". When I use "jupyter qtconsole --kernel octave" to test octave_kernel, what I obtained is shown as

```
Jupyter QtConsole 5.3.0
Octave Kernel v0.34.2 running GNU Octave v[?2004h

In[1]: 1+1
[?2004l
[?2004h
[?2004l
error: graphics_toolkit: 2004h toolkit is not available
error: called from
    graphics_toolkit at line 84 column 5
[?2004h
[?2004l
[?2004h
s = 2
[?2004h

In[2]: a=1
a = 104l
[?2004h
```

There are many weird "[?2004h" and "[?2004l" in the output, and I think the problem that graphics_toolkit can't initialize correctly is also caused by these weird "2004h". Also, the output of "a=1" is "a=104l" which seems to be a complete of a hex-number. These problems also appears in "jupyter console --kernel octave" and jupyter-lab. 

As for the platform, this problem occurs in both Manjaro-Gnome-zsh and Arch-Mate-bash where the version of Octave is 7.1.0, but everything goes well on my openSUSE Leap 15.3 with Octave-5.3.0. Besides, if I install Octave 6.3.0 on Manjaro by conda-forge, this problem also disappears although there are some other problems.

What is the reason of this problem? Is it a problem with Arch Linux or Octave 7.1.0?

**Comment by ThomasUCAS** [comment]
> 

Great! I think so, and I notice that you mentioned another issue https://github.com/blink1073/oct2py/issues/237 which is actually the same as my problem although that issue is about "oct2py" rather than "octave_kernel". 

But I'm still confused why this bug occurs in Octave 7.1.0. Actually, I tried Octave 6.3.0, 6.4.0 and 7.1.0 from conda-forge after I submitted this issue, and this bug only occurs in Octave 7.1.0. So I think this bug is probably irrelevant with ArchLinux or Manjaro and also irrelevant with bash or zsh, but depends on the version of Octave. Does it mean that "pexpect" is compatible with Octave 6.3.0 and 6.4.0 but not with 7.1.0?
